docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+  - The templates in the `fxa-auth-server` do not contain heavy business logic so building them with a frontend-heavy framework like React might complicate things
+  - Since we will be converting React components into static HTML pages, another layer of complexity as well as new dependencies would be added to the project
+
+### Option C - Use EJS
+
+- Description
+  - Template the auth server mail components using [EJS](https://ejs.co/)
+- Pros
+  - The syntax and logic of EJS is similar to JavaScript, leading to a very small learning curve.
+  - EJS has a large community of users which will ensure faster help and support.
+  - It is a simple templating engine and far easier to use as its plain Javascript and since we want to keep our mail templates straightforward, using EJS could be a good choice.
+- Cons
+  - Adds a new dependency to the stack
+  - Lacks support for `block` functionality, which will make resuing components a bit difficult.
+  - Although we can implement `block` functionality with the help of third-party support, most of the projects are outdated and not maintained anymore.
+
+### Pros and Cons of using MJML as part of our templating stack
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Along with the framework, we considered using [MJML](https://mjml.io/) as our potential markup language, designed to reduce the pain of coding a responsive email.
+
+- Pros
+  - MJML is a component-based library having its own pre-built components as well as gives us the ability to define custom compononents which translates to responsive HTML after compilation. These components can be reused across the templates and modified by passing parameters e.g. [this social links component](https://documentation.mjml.io/#mj-social)
+  - MJML can help us create responsive emails using mobile-first approach and it controls that responsiveness automatically needing very less to no developer intervention.
+  - This library has been created by keeping in mind all the industry standards as well as the quirks of old email clients, so we will eventually be saving time on investigating cross-browser compatability of our templates
+  - MJML has by far the highest weekly downloads and github stars in comparision to its alternatives like [Foundation for Emails](https://github.com/foundation/foundation-emails).
+  - MJML can be integrated with React to generate responsive emails and there is also a [library](https://github.com/wix-incubator/mjml-react) that could help us with that purpose.
+- Cons
+  - Adds a new dependency to the stack
+  - Creates a slight learning curve
+  - While we may not have much component reuse between other packages that use React anyway, using MJML in email components would make it more challenging to reuse them elsewhere
```

Comment:
Given the different nature of email rendering vs. typical browser-based rendering, do we expect email components to be used outside of emails?

Reply:
I feel like there is potential reuse especially since classes can be given defaults and overwritten by a parent component passing them down as a prop, but yeah I don't think there will be a ton of it.